website/client/src/pages/settings/promoCode.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+  <div class="row standard-page">
+    <div class="col-12">
+      <h1
+        v-once
+        class="page-header"
+      >
+        {{ $t('promoCode') }}
+      </h1>
+
+      <div class="input-area">
+        <div
+          class="form-inline"
+          role="form"
+        >
+          <input
+            v-model="couponCode"
+            class="form-control"
+            type="text"
+            :placeholder="$t('promoPlaceholder')"
+          >
+        </div>
+        <div
+          v-once
+          class="small mt-2"
+        >
+          {{ $t('couponText') }}
+        </div>
+        <save-cancel-buttons
+          :hide-cancel="true"
+          primary-button-label="submit"
+          @saveClicked="enterCoupon()"
+        />
+        <div v-if="canCreateCoupons">
+          <hr>
+          <h4>{{ $t('generateCodes') }}</h4>
+          <div
+            class="form"
+            role="form"
+          >
+            <div class="form-group">
+              <input
+                v-model="codes.event"
+                class="form-control"
+                type="text"
+                placeholder="Event code (eg, 'wondercon')"
+              >
+            </div>
+            <div class="form-group">
+              <input
+                v-model="codes.count"
+                class="form-control"
+                type="number"
+                placeholder="Number of codes to generate (eg, 250)"
+              >
+            </div>
+            <div class="form-group">
+              <button
+                class="btn btn-primary"
+                type="submit"
+                @click="generateCodes(codes)"
+              >
+                {{ $t('generate') }}
+              </button>
+              <a
+                class="btn btn-secondary"
+                :href="getCodesUrl"
+              >{{ $t('getCodes') }}</a>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<style scoped>
+  .form-control {
+    width: 100%;
```

Comment:
We have the `w-100` helper class in Bootstrap that can cover this!

Reply:
can clean those up - I just moved it from the old folder to the new and git just thought "yeah I'll mark it as deleted and created from scratch" ... thanks git ^^